src/test/regress/expected/distributed_functions.out

```diff
@@ -106,6 +110,170 @@ SELECT * FROM run_command_on_workers('SELECT function_tests.add(2,3);') ORDER BY
+ERROR:  unsupported ALTER FUNCTION ... SET ... FROM CURRENT for a distributed function
+DETAIL:  SET FROM CURRENT is not supported for distributed functions, instead use the SET ... TO ... syntax with a constant value.
+SELECT public.verify_function_is_same_on_workers('function_tests.add(int,int)');
+ verify_function_is_same_on_workers 
+------------------------------------
+ t
+(1 row)
+
+ALTER FUNCTION add(int,int) SET client_min_messages FROM CURRENT SECURITY DEFINER;
+ERROR:  unsupported ALTER FUNCTION ... SET ... FROM CURRENT for a distributed function
+DETAIL:  SET FROM CURRENT is not supported for distributed functions, instead use the SET ... TO ... syntax with a constant value.
+SELECT public.verify_function_is_same_on_workers('function_tests.add(int,int)');
+ verify_function_is_same_on_workers 
+------------------------------------
+ t
+(1 row)
+
+-- rename function and make sure the new name can be used on the workers while the old name can't
+ALTER FUNCTION add(int,int) RENAME TO add2;
+SELECT public.verify_function_is_same_on_workers('function_tests.add2(int,int)');
+ verify_function_is_same_on_workers 
+------------------------------------
+ t
+(1 row)
+
+SELECT * FROM run_command_on_workers('SELECT function_tests.add(2,3);') ORDER BY 1,2;
+ nodename  | nodeport | success |                                result                                
+-----------+----------+---------+----------------------------------------------------------------------
+ localhost |    57637 | f       | ERROR:  function function_tests.add(integer, integer) does not exist
+ localhost |    57638 | f       | ERROR:  function function_tests.add(integer, integer) does not exist
+(2 rows)
+
+SELECT * FROM run_command_on_workers('SELECT function_tests.add2(2,3);') ORDER BY 1,2;
+ nodename  | nodeport | success | result 
+-----------+----------+---------+--------
+ localhost |    57637 | t       | 5
+ localhost |    57638 | t       | 5
+(2 rows)
+
+ALTER FUNCTION add2(int,int) RENAME TO add;
+-- change the owner of the function and verify the owner has been changed on the workers
+ALTER FUNCTION add(int,int) OWNER TO functionuser;
+SELECT public.verify_function_is_same_on_workers('function_tests.add(int,int)');
+ verify_function_is_same_on_workers 
+------------------------------------
+ t
+(1 row)
+
+SELECT run_command_on_workers($$
+SELECT row(usename, nspname, proname)
+FROM pg_proc
+JOIN pg_user ON (usesysid = proowner)
+JOIN pg_namespace ON (pg_namespace.oid = pronamespace)
+WHERE proname = 'add';
+$$);
+                 run_command_on_workers                  
+---------------------------------------------------------
+ (localhost,57637,t,"(functionuser,function_tests,add)")
+ (localhost,57638,t,"(functionuser,function_tests,add)")
+(2 rows)
+
+-- change the schema of the function and verify the old schema doesn't exist anymore while
+-- the new schema has the function.
+ALTER FUNCTION add(int,int) SET SCHEMA function_tests2;
```

Comment:
Should we not create `function_tests2 ` above and let the dependency logic create it? This is also fine, but that looks fancier :)

Reply:
The schema is actually created on the workers via the dependency logic at this point in time. We do create the schema on the coordinator at the top of the file. Having it here might make it more clear it is propagated to the workers here, however it is hard to read which schema's a test uses if they are not all created at the top, hence the `CREATE SCHEMA` statement all the way at the top